lib/cloudini.ex

```diff
@@ -45,19 +45,19 @@ defmodule Cloudini do
   end
 
   def version(client) do
-    with_metrics "version" do
+    with_metrics do
       ClientAPI.version(client)
     end
   end
 
   def upload_image(client, path, opts \\ []) do
-    with_metrics "upload_image" do
+    with_metrics do
```

Comment:
I initially thought that we won't mention metrics at all in this library and instead leave this up to the client. I think it's a good idea for clients to wrap external library anyway (so that they can switch to something else, replace in tests, add metrics, add logging, add retrying logic etc):

```elixir
defmodule MyApp.ImageUploader do
  import MyApp.Metrics

  def upload_image(path, opts \\ []) do
    with_metrics do
      Cloudini.upload_image(client, path, opts)
    end
  end

  def client do
    Cloudini.new
  end
end
```

I think exposing instrumentation hooks is useful when library exposes behaviours that the consumers are implementing (e.g. consumer implements a phoenix controller) and not that much when the consumer is just using library directly. But yeah, in our particular case where we control both codebases, if there are hooks we wouldn't need to re-implement public interface of Cloudini just to use metrics. 

WDYT?